src/components/Chart/resolveOptions.js

```diff
@@ -92,15 +105,54 @@ export default function resolveOptions(conditions) {
             hover: {
                 animationDuration: 0,
             },
-            responsiveAnimationDuration: 0,
         };
     }
+
+    if (showLabelsOnBars) {
+        options = {
+            ...options,
+            hover: {
+                animationDuration: disableAnimations
+                    ? 0
+                    : ChartJS.defaults.global.hover.animationDuration,
+            },
+            animation: {
+                duration: disableAnimations ? 0 : ChartJS.defaults.global.animation.duration,
+                onComplete:
+                    showLabelsOnBars && type === 'bar'
+                        ? function() {
+                              const chart = this.chart;
+                              const ctx = chart.ctx;
+
+                              ctx.font = ChartJS.helpers.fontString(
+                                  ChartJS.defaults.global.defaultFontSize,
+                                  ChartJS.defaults.global.defaultFontStyle,
+                                  ChartJS.defaults.global.defaultFontFamily,
+                              );
+                              ctx.fillStyle = legend.label;
+                              ctx.textAlign = 'center';
+                              ctx.textBaseline = 'bottom';
+
+                              this.data.datasets.forEach((dataset, i) => {
+                                  const meta = chart.controller.getDatasetMeta(i);
+                                  meta.data.forEach((bar, index) => {
+                                      const dataValue = dataset.data[index];
+                                      ctx.fillText(dataValue, bar._model.x, bar._model.y - 5);
+                                  });
+                              });
+                          }
+                        : null,
```

Comment:
we should move this complex function to another file in order to test it in isolation